src/lib/Preview.js

```diff
@@ -779,35 +834,33 @@ class Preview extends EventEmitter {
         }
 
         try {
-            // Save reference to the file and update logger
+            // Set current file to file data from server and update file in logger
             this.file = file;
             this.logger.setFile(file);
 
             // Keep reference to previously cached file version
             const cachedFile = this.cache.get(file.id);
 
             // Explicitly uncache watermarked files, otherwise update cache
-            const isWatermarked = file.watermark_info && file.watermark_info.is_watermarked;
-            if (isWatermarked) {
+            const isFileWatermarked = isWatermarked(file);
+            if (isFileWatermarked) {
```

Comment:
Unless you're using `isFileWatermarked` elsewhere, could you just put the `isWatermarked()` call into the if?

Reply:
Used a few lines below